src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -1,11 +1,12 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
-public interface MachineProducer {
+public interface MachineProducer<T extends Machine> {
     /**
-     * In your implementations - create 2-3 models, add them to the list and return
+     ** In your implementations - create 2-3 models, add them to the list and return
      * @return - the list of models
      */
-    List<Object> get();
+    List<T> get();
```

Comment:
Do not use raw types. Always parametrize interfaces and classes if it is possible. Here, you have correctly parameterized the `MachineProducer` interface to ensure that only `Machine` types or its subtypes can be used, which is in line with the checklist requirements.